aptos-move/block-executor/src/limit_processor.rs

```diff
@@ -83,6 +89,32 @@ impl<T: Transaction> BlockGasLimitProcessor<T> {
         }
     }
 
+    pub(crate) fn recompute_acc_eff_block_gas_on_module_rw_conflict(&mut self) {
+        if !self
+            .block_gas_limit_type
+            .use_module_publishing_block_conflict()
```

Comment:
let's move this out - to not even compute/check for module conflicts if this flag is off (similarly how we don't compute ReadWriteSummary if corresponding flag is off)